src/3d/qgstessellator.cpp

```diff
@@ -282,6 +282,21 @@ static QgsPolygon *_transform_polygon_to_new_base( const QgsPolygon &polygon, co
 
 static bool _check_intersecting_rings( const QgsPolygon &polygon )
 {
+  QList<QgsGeometry> geomRings;
+  geomRings << QgsGeometry( polygon.exteriorRing()->clone() );
+  for ( int i = 0; i < polygon.numInteriorRings(); ++i )
+    geomRings << QgsGeometry( polygon.interiorRing( i )->clone() );
+
+  // we need to make sure that the polygon has no rings with self-intersection: that may
+  // crash the tessellator. The original geometry maybe have been valid and the self-intersection
+  // was introduced when transforming to a new base (in a rare case when all points are not in the same plane)
+
+  for ( int i = 0; i < geomRings.count(); ++i )
+  {
+    if ( !geomRings[i].isSimple() )
+      return false;
```

Comment:
Couldn't this check already be done in the above loop to avoid looping twice? Extra points for an implementation without index based access ;)

Reply:
That's possible but won't really help us much - the assumption is that 99.9% of time the rings will be simple and the extra loop is not adding much overhead (compared to isSimple() call or cloning of rings which should be optimized away).

(read: I am just being to lazy to go back, update the code, recompile, wait for travis again)